sympy/ntheory/factor_.py

```diff
@@ -1482,3 +1482,73 @@ def eval(cls, n, k=1):
             else:
                 return Mul(*[(p**(k*(e + 1)) - 1)/(p**k - 1) if k != 0
                            else e + 1 for p, e in factorint(n).items()])
+
+
+class core(Function):
+    """
+    Calculate core(n,t) = `core_t(n)` of a positive integer n
+
+    ``core_2(n)`` is equal to the squarefree part of n
+
+    If n's prime factorization is:
+
+    .. math ::
+        n = \prod_{i=1}^\omega p_i^{m_i},
+
+    then
+
+    .. math ::
+        core_t(n) = \prod_{i=1}^\omega p_i^{m_i \mod t}.
+
+    Parameters
+    ==========
+
+    t : core(n,t) calculates the t-th power free part of n
+
+        ``core(n, 2)`` is the squarefree part of ``n``
+        ``core(n, 3)`` is the cubefree part of ``n``
+
+        Default for t is 2.
+
+    References
+    ==========
+
+    .. [1] http://en.wikipedia.org/wiki/Square-free_integer#Squarefree_core
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import core
+    >>> core(24, 2)
+    6
+    >>> core(9424,3)
+    1178
+    >>> core(379238)
+    379238
+    >>> core(15**11, 10)
+    15
+
+    See Also
+    ========
+
+    factorint
+    """
+
+    @classmethod
+    def eval(cls, n, t=2):
+        n = sympify(n)
```

Comment:
for speed and simplicity we generally handle this as
```python
n = as_int(n)
t = as_int(t)
```
then
```
if sympify(n).is_prime:
    return n
if n <= 0:
    er
elif t <= 1:
    er
else:
    calc
```

`as_int` will handle raising the error.